src/allmydata/web/status.py

```diff
@@ -1551,6 +1552,31 @@ def render_JSON(self, req):
         req.setHeader("content-type", "text/plain")
         return json.dumps(stats, indent=1) + "\n"
 
+    @render_exception
+    def render_OPENMETRICS(self, req):
+        req.setHeader("content-type", "application/openmetrics-text; version=1.0.0; charset=utf-8")
+        stats = self._provider.get_stats()
+        ret = u""
```

Comment:
Practically, it's *probably* fine to build up `ret` using repeated concatenation but most folks will prefer to build a list of lines and then join it at the end.  CPython has an optimization in place specifically to handle this case which makes repeated concatenation O(N) instead of O(N^2) but there's no guarantee about exactly which cases it will catch and I don't remember if other runtimes like PyPy have copied it or not.

Reply:
Thanks!